src/lib/types.ts

```diff
@@ -20,11 +20,22 @@ export type Edge = {
 
 export type LearningUnit = {
 	id: string;
+	children: LearningUnit[];
 	mediaTime?: number;
 	words?: number;
 	requiredSkills: Skill[];
 	teachingGoals: Skill[];
 	suggestedSkills: { weight: number; skill: Skill }[];
+	externalRequiredSkills?: Skill[];
+	exportedRequiredSkills?: Skill[];
+	unusedRequiredSkills?: Skill[];
+	taughtSkills?: Skill[];
+	externalRecommendedSkills?: { weight: number; skill: Skill }[];
+	exportedRecommendedSkills?: { weight: number; skill: Skill }[];
+	unusedRecommendedSkills?: { weight: number; skill: Skill }[];
+	getTeachingGoals(): Skill[];
+	getRequiredSkills(): Skill[];
+	getSuggestedSkills(): { weight: number; skill: Skill }[];
```

Comment:
I'm not fully acquainted with all the coding conventions employed within the skill-lib, but it appears that there's a general preference for functional programming over object-oriented programming. Given this context, the type definitions should primarily outline data properties rather than behaviors.